component/rules.jsonnet

```diff
@@ -0,0 +1,132 @@
+                // Attach Signalilo/Icinga heartbeat labeling to Watchdog rule.
+                if rule.alert == 'Watchdog' then
+                  rule {
+                    labels: {
+                      heartbeat: '60s',
+                      severity: 'critical',
+                    },
+                    annotations+: {
+                      description: 'This is a dead mans switch meant to ensure that the entire alerting pipeline is functional.',
+                      summary: 'Alerting dead mans switch',
+                    },
+                  }
+                else
+                  rule,
+              group.rules
+            ),
+          }
+        else
+          group,
+      super.groups
+    ),
+  },
+};
+
+local ruleFilter(group) =
+  local internalRuleFilter =
+    if group.name == 'kubernetes-apps' then
+      [ 'KubeHpaMaxedOut', 'KubeHpaReplicasMismatch' ]
+    else if group.name == 'kubernetes-resources' then
+      [ 'CPUThrottlingHigh' ]
+    else if group.name == 'kubernetes-system-apiserver' then
+      // This can be removed once https://github.com/coreos/kube-prometheus/pull/516 got merged
+      [ 'AggregatedAPIDown' ]
+    else
+      [];
+  local userRuleFilter =
+    [ r.name for r in exclude_rules if r.group == group.name ];
+
+  // merge user-provided and hard-coded rule names to filter out
+  std.set(internalRuleFilter + userRuleFilter);
+
+local filterRules = {
+  prometheusAlerts+:: {
+    groups: std.map(
+      function(group)
+        group {
+          rules: std.filter(
+            function(rule)
+              !std.member(ruleFilter(group), rule.alert),
+            group.rules
+          ),
+        },
+      super.groups
+    ),
+  },
+};
+
+local additionalRules = {
+  prometheusAlerts+:: {
+    groups+: [
+      {
+        name: 'node-utilization',
+        rules: [
+          {
+            alert: 'node_cpu_load5',
+            expr: 'max by(instance) (node_load5) / count by(instance) (node_cpu_seconds_total{mode="idle"}) > 2',
+            'for': '30m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              message: '{{$labels.instance}}: Load higher than (current value is: {{ $value }})',
+            },
+          },
+          {
+            alert: 'node_memory_free_percent',
+            expr: '(node_memory_MemTotal_bytes - node_memory_MemAvailable_bytes) / node_memory_MemTotal_bytes > 0.97',
+            'for': '30m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              message: '{{$labels.node}}: Memory usage more than 97% (current value is: {{ $value | humanizePercentage }})%',
+            },
+          },
+        ],
+      },
+    ],
+  },
+};
+
+local kp =
+  (import 'kube-prometheus/kube-prometheus.libsonnet') +
+  (import 'kube-prometheus/kube-prometheus-managed-cluster.libsonnet') +
+  filterRules +
+  alterRules +
+  additionalRules +
```

Comment:
Adding `additionalRules` last, makes this hard coded. `filterRules` and `alterRules` can not influence it. Is this intentional?

Reply:
Adjusted, thanks for pointing that out.

(side note: alterRules probably shouldn't be used to modify rules which are defined in the same jsonnet file, but that's another discussion)